src/compiler/checker.ts

```diff
@@ -8839,13 +8840,36 @@ namespace ts {
             })!.parent;
         }
 
+        function getInstanceTypeOfClassSymbol(classSymbol: Symbol): Type {
+            const classType = getDeclaredTypeOfSymbol(classSymbol) as GenericType;

```

Comment:
You should *probably* check that this cast is safe - namely by checking that the object flags of the type contain the class/interface object flag. Especially since other kinds of types can have type parameters (eg, mapped types).